python/pyspark/context.py

```diff
@@ -156,7 +156,7 @@ def _do_init(self, master, appName, sparkHome, pyFiles, environment, batchSize,
         self.sparkHome = self._conf.get("spark.home", None)
 
         # Let YARN know it's a pyspark app, so it distributes needed libraries.
-        if self.master == "yarn-client":
+        if self.master == "yarn":
```

Comment:
```
./dev/change-scala-version.sh 2.110
mvn -Pyarn -Phadoop-2.4 -Dscala-2.10 -DskipTests clean package
```

Please take a try. Also be noted repl behavior is different between 2.11 and 2.10.